addons/l10n_it_edi_sdicoop/models/account_edi_format.py

```diff
@@ -224,28 +233,31 @@ def _l10n_it_post_invoices_step_2(self, invoices):
                     During the following 15 days, the Exchange System will try to forward the FatturaPA\
                     file to the Administration in question again.'),
                 }
-            elif state == 'notificaEsito':
+            elif state == 'notificaEsito' and response.get('file'):
                 outcome = response_tree.find('Esito').text
                 if outcome == 'EC01':
-                    to_return[invoice] = {'attachment': invoice.l10n_it_edi_attachment_id}
+                    to_return[invoice] = {'attachment': invoice.l10n_it_edi_attachment_id, 'success': True}
                 else:  # ECO2
                     to_return[invoice] = {'error': _('The invoice was refused by the addressee.'), 'blocking_level': 'error'}
             elif state == 'NotificaDecorrenzaTermini':
-                to_return[invoice] = {'error': _('Expiration of the maximum term for communication of acceptance/refusal'), 'blocking_level': 'error'}
-            proxy_acks.append(id_transaction)
+                to_return[invoice] = {'attachment': invoice.l10n_it_edi_attachment_id, 'success': True}
 
-        try:
-            proxy_user._make_request(server_url + '/api/l10n_it_edi/1/ack',
-                                     params={'transaction_ids': proxy_acks})
-        except AccountEdiProxyError as e:
-            # Will be ignored and acked again next time.
-            _logger.error('Error while acking file to SdiCoop: %s', e)
+            if response.get('file'):
+                proxy_acks.append(id_transaction)
```

Comment:
Why is this not done in the first `if response.get('file'):` 20ish lines above?

Reply:
It is true that if there is no new message, there is in principle also nothing to change to the error information.  It is only that if something would not be synchronized, it could synchronize in the cases where you do not need the contents of the file, ...